### ISSUE TITLE
Flag "--log warn" hides "err"-level messages

### ISSUE THREAD

**Issue by jgarren** [jspm/jspm-cli#1354]
In `lib/ui.js` we have:
```js
var logTypes = ['err', 'warn', 'ok', 'info', 'debug'];
// ...
exports.log = function(type, msg) {
  // ...
  if ((logTypes.indexOf(type) || 3) <= ui.logLevel) {
```

Suppose `type` is "err". Then `logTypes.indexOf(type)` is 0. But `0 || 3` is 3. So, if you had set the log level to "warn", then `ui.logLevel` is 1, but `3 <= 1` is of course false, so the message doesn't get logged.

I think we've all been bitten by the old `0 || something` evaluation before.

One of many possible solutions:
```js
exports.log = function(type, msg) {
  // ...
  var logLevel = logTypes.indexOf(type)
  logLevel = logLevel === -1 ? 3 : logLevel
  if (logLevel <= ui.logLevel) {
```

**Comment by guybedford** [comment]
@jgarren thanks, yes this is a bug. I fixed it in the 0.17 branch in https://github.com/jspm/jspm-cli/blob/0.17/lib/ui.js#L109, but it hasn't been backported yet.

**Comment by guybedford** [comment]
Fixed in 0.16.16.
